### PR TITLE
code update and improvements

### PR DESCRIPTION
**Pull request by solegalli**

### CODE REVIEW

**Comment by solegalli** [comment]
@ChristopherGS I am fixing docstrings and pushing to same branch, sorry, not elegant.
Also, in commit "fixed docstrings in imputers/encoders", I added one additional parameter check to RareLabelEncoder, and change the test to accommodate.

**Comment by solegalli** [comment]
@ChristopherGS I am on fire with git :DDDD

File: requirements.txt [hunk 1]
```diff
@@ -1,6 +1,5 @@
-numpy>=1.12.1
-pandas>=0.23.0
-scikit-learn>=0.18.1
-scipy>=0.19.0
-statsmodels>=0.8.0
-numpydoc>=0.6.0
```
Comment (ChristopherGS) [inline requirements.txt:6]: @solegalli do you need `numpydoc`?

File: requirements.txt [hunk 2]
```diff
@@ -1,6 +1,5 @@
...
+numpy>=1.18.2,<1.19.0
```
Comment (ChristopherGS) [inline requirements.txt:1]: pinning version ranges to avoid unexpected breaking changes

File: feature_engine/variable_manipulation.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
+# Authors: Soledad Galli <solegalli@protonmail.com>
```
Comment (ChristopherGS) [inline feature_engine/variable_manipulation.py:1]: extracted and renamed from utils

File: feature_engine/tests/test_categorical_encoder.py [hunk 4]
```diff
@@ -1,252 +0,0 @@
-# Authors: Soledad Galli <solegalli1@gmail.com>
```
Comment (ChristopherGS) [inline feature_engine/tests/test_categorical_encoder.py:1]: moved tests directory up one level

File: .circleci/config.yml [hunk 5]
```diff
@@ -1,23 +1,51 @@
 version: 2
 jobs:
```
Comment (ChristopherGS) [inline .circleci/config.yml:2]: add python version testing and tox

File: README.md [hunk 6]
```diff
@@ -71,6 +71,16 @@ data_encoded = rare_encoder.transform(data)
 
 See more usage examples in the jupyter notebooks in the example folder of this repository, or in the documentation: http://feature-engine.readthedocs.io
 
+### Contributing
```
Comment (ChristopherGS) [inline README.md:74]: modest addition to provide contribution basics

File: feature_engine/__init__.py [hunk 7]
```diff
@@ -1 +1,10 @@
+import pathlib
```
Comment (ChristopherGS) [inline feature_engine/__init__.py:1]: nice

File: test_requirements.txt [hunk 8]
```diff
@@ -1 +1,3 @@
-pytest>=4.3.1,<6.0
\ No newline at end of file
+-r requirements.txt
+pytest>=5.4.1,<6.0
+flake8>=3.7.9,<3.8.0
```
Comment (ChristopherGS) [inline test_requirements.txt:3]: added flake8 for stylechecks

File: tox.ini [hunk 9]
```diff
@@ -0,0 +1,33 @@
+[tox]
```
Comment (ChristopherGS) [inline tox.ini:1]: add tox

File: requirements.txt (hunk 1 above)
Comment (solegalli) [reply requirements.txt:6]: No just for the docs, but I have that in a separate req file inside the doc folder. I was going to commit req files last. But you took care of it. Thanks a lot. Cheers
Sole

File: tox.ini [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+deps =
+    -rtest_requirements.txt
+
+[testenv:stylechecks]
+envdir = {toxworkdir}/unit_tests
+
+deps =
+    flake8
+
+# TODO: apply linting to other modules
```
Comment (ChristopherGS) [inline tox.ini:33]: note the todo, currently stylechecks just run against one file.

File: feature_engine/dataframe_checks.py [hunk 11]
```diff
@@ -0,0 +1,26 @@
+# Authors: Soledad Galli <solegalli@protonmail.com>
```
Comment (ChristopherGS) [inline feature_engine/dataframe_checks.py:1]: extracted and renamed from utils

**Pull request merged**